crates/volta-core/src/tool/npm.rs

```diff
@@ -26,25 +28,33 @@ where
     }
 }
 
-fn is_global_npm_install() -> bool {
-    let command = args_os()
-        .skip(1)
-        .skip_while(|arg| match arg.to_str() {
-            Some(arg) => arg.starts_with("-"),
-            None => false,
-        })
-        .next();
+fn check_npm_install() -> CommandArg {
+    // npm global installs will have `-g` or `--global` somewhere in the
+    // argument list
+    let is_global =
+        args_os().any(|arg| arg == OsString::from("-g") || arg == OsString::from("--global"));
```

Comment:
It may be a premature optimization, but at this point you could check `is_global` and short-circuit the argument filtering and comparison if it's not a global command, instead of doing all of that and then checking `is_global`.

Reply:
Ah, good point.

Reply:
Resolved in 18856cdb; thanks!